runtimeconfig/google/cloud/runtimeconfig/config.py

```diff
@@ -0,0 +1,291 @@
+
+        Example:
+        ``projects/my-project/configs/my-config``
+
+        :rtype: string
+        :returns: The full name based on project and config names.
+        """
+        if not self.name:
+            raise ValueError('Missing config name.')
+        return 'projects/%s/configs/%s' % (self._client.project, self.name)
+
+    @property
+    def path(self):
+        """URL path for the config's APIs.
+
+        :rtype: string
+        :returns: The URL path based on project and config names.
+        """
+        return '/%s' % (self.full_name)
+
+    def variable(self, variable_name):
+        """Factory constructor for variable object.
+
+        .. note::
+          This will not make an HTTP request; it simply instantiates
+          a variable object owned by this config.
+
+        :type variable_name: string
+        :param variable_name: The name of the variable to be instantiated.
+
+        :rtype: :class:`google.cloud.runtimeconfig.variable.Variable`
+        :returns: The variable object created.
+        """
+        return Variable(name=variable_name, config=self)
+
+    def _require_client(self, client):
+        """Check client or verify over-ride.
+
+        :type client: :class:`google.cloud.runtimconfig.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current zone.
+
+        :rtype: :class:`google.cloud.runtimeconfig.client.Client`
+        :returns: The client passed in or the currently bound client.
+        """
+        if client is None:
+            client = self._client
+        return client
+
+    def _set_properties(self, api_response):
+        """Update properties from resource in body of ``api_response``
+
+        :type api_response: httplib2.Response
+        :param api_response: response returned from an API call
+        """
+        self._properties.clear()
+        cleaned = api_response.copy()
+        if 'name' in cleaned:
+            self.name = _config_name_from_full_name(cleaned.pop('name'))
+        self._properties.update(cleaned)
+
+    def exists(self, client=None):
+        """Determines whether or not this config exists.
+
+        :type client: :class:`~google.cloud.runtimeconfig.client.Client` or
+                      ``NoneType``
+        :param client: Optional. The client to use.  If not passed, falls back
+                       to the ``client`` stored on the current config.
+
+        :rtype: boolean
+        :returns: True if the config exists in Cloud Runtime Configurator.
+        """
+        client = self._require_client(client)
+        try:
+            # We only need the status code (200 or not) so we seek to
+            # minimize the returned payload.
+            query_params = {'fields': 'name'}
+            # We intentionally pass `_target_object=None` since fields=name
+            # would limit the local properties.
```

Comment:
`_target_object` is only relevant to the storage Batch implementation, please remove this comment and remove `_target_object=None` below